extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoServiceBindingConverter.java

```diff
@@ -0,0 +1,104 @@
+package io.quarkus.mongodb.runtime;
+
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBinding;
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConfigSource;
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConverter;
+import org.jboss.logging.Logger;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.lang.String.format;
+
+public class MongoServiceBindingConverter implements ServiceBindingConverter  {
+    private static final Logger LOGGER = Logger.getLogger(MongoServiceBindingConverter.class);
+
+    private static final String BINDING_TYPE = "MongoDB";
+    private static final String BINDING_CONFIG_SOURCE_NAME = BINDING_TYPE + "-k8s-service-binding-source";
+    private static final String MONGO_DB_CONNECTION_STRING = "MONGODB_CONNECTION_STRING";
+
+    // DB properties
+    private static final String DB_USER = "username";
+    private static final String DB_PASSWORD = "password";
+    private static final String DB_HOST = "host";
+    private static final String DB_PORT = "port";
+    private static final String DB_DATABASE = "database";
+    private static final String DB_PREFIX = "mongodb+srv://";
+    private static final String DB_DEFAULT_OPTIONS = "?retryWrites=true&w=majority";
+
+    // Quarkus properties
+    private static final String QUARKUS_DS_USERNAME = "quarkus.datasource.username";
+    private static final String QUARKUS_DS_PASSWORD = "quarkus.datasource.password";
```

Comment:
These properties are specific to relational database handling and should not be set at all for Mongo. The Mongo integration adds these to the connection string, see https://quarkus.io/guides/all-config#quarkus-mongodb-client_quarkus.mongodb.connection-string